#! /bin/sh /usr/share/dpatch/dpatch-run
## 0012-upstream-allow-report-above-4GB-mboxsize by Sven Mueller <debian@incase.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Upstream patch to allow reporting of mailbox sizes above 4GB

@DPATCH@
--- a/imap/annotate.c
+++ b/imap/annotate.c
@@ -703,7 +703,13 @@ static void annotation_get_size(const ch
     struct mailbox mailbox;
     struct index_record record;
     int r = 0, msg;
+#ifdef HAVE_LONG_LONG_INT
+    unsigned long long totsize = 0;
+# define SIZE_FMT "%llu"
+#else 
     unsigned long totsize = 0;
+# define SIZE_FMT "%lu"
+#endif
     char value[21];
     struct annotation_data attrib;
 
@@ -735,7 +741,7 @@ static void annotation_get_size(const ch
 
     mailbox_close(&mailbox);
 
-    if (r || snprintf(value, sizeof(value), "%lu", totsize) == -1)
+    if (r || snprintf(value, sizeof(value), SIZE_FMT, totsize) == -1)
 	return;
 
     memset(&attrib, 0, sizeof(attrib));
